dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/SmeDataApi.kt

```diff
@@ -0,0 +1,45 @@
+package org.dataland.datalandbackend.api
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.responses.ApiResponses
+import io.swagger.v3.oas.annotations.security.SecurityRequirement
+import org.dataland.datalandbackend.frameworks.sme.model.SmeData
+import org.dataland.datalandbackend.model.companies.CompanyAssociatedData
+import org.dataland.datalandbackend.model.metainformation.DataMetaInformation
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestPart
+import org.springframework.web.multipart.MultipartFile
+
+/**
+ * Defines the restful dataland-backend API regarding private sme company data.
+ */
+@RequestMapping("/data/sme")
+@SecurityRequirement(name = "default-bearer-auth")
+@SecurityRequirement(name = "default-oauth")
+interface SmeDataApi {
```

Comment:
This interface causes a code smell because it holds only one method and sonar wants it to be changed into a "fun interface". This however causes an error with the default value of the multipart endpoint. As we plan to introduce the get endpoints in the next story i would vote for ignoring the code smell for now.